requirements.txt

```diff
@@ -1,22 +1,17 @@
 # spam-filter doesn't work without babel (but somehow doesn't list it in its requirements)
-Trac[pygments, babel]==1.4.4
+Trac[pygments, babel]==1.6.0
 dnspython==1.15
-spambayes == 1.1b1
-psycopg2==2.7.6.1 --no-binary=psycopg2
-docutils==0.14
+psycopg2==2.9.9 --no-binary=psycopg2
 Django==1.11.29
-libsass==0.17.0
-Genshi==0.7.7  # still required by some plugins
+libsass==0.23.0
 
 # Trac plugins
-https://trac.edgewall.org/browser/plugins/1.4/spam-filter?format=zip
-# TracXMLRPC from PyPI does not (yet) have a 1.2.0 release (compatible with Trac 1.4)
+https://trac.edgewall.org/browser/plugins/trunk/spam-filter?rev=17752&format=zip
+# TracXMLRPC from PyPI does not (yet) have a 1.2.0 release (compatible with Trac >=1.4)
 https://trac-hacks.org/browser/xmlrpcplugin/trunk?rev=18591&format=zip
 
-oauthlib==2.1.0
-requests==2.20.1
-requests-oauthlib==1.0.0
```

Comment:
Are these 3 packages required even without bumping to the Trac 1.6?

Reply:
They're required for the "log in with github" functionality provided by `trac-github`.I was able to remove the lines from the requirements here because I'm using the `[oauth]` extra of `trac-github[oauth]` which install them automatically.

We could make this change to our requirements file already without updating Trac, yes, but it doesn't (shouldn't?) have any effect in practice because the dependencies are installed anyway. It only serves to reduce the size of the requirements file.